CMakeLists.txt

```diff
@@ -197,6 +197,9 @@ cmake_policy(SET CMP0042 NEW)   #MACOSX_RPATH is enabled by default
 #Windows 10 camplains 
 cmake_policy(SET CMP0054 NEW)   # quoted arguments will not be further dereferenced
 
+#OSX complains 
+cmake_policy(SET CMP0087 NEW)   # evaluate generator expressions for install(CODE) and install(SCRIPT)
```

Comment:
Policy CMP0078 was defined in CMake version 3.13. As the minimum required version for building lomse is 2.8.10, any version lower than 3.13 generates an error:
```
CMake Error at CMakeLists.txt:208 (cmake_policy):
  Policy "CMP0087" is not known to this version of CMake.
```
Please change to:

```
if (${CMAKE_VERSION} GREATER "3.12")
    cmake_policy(SET CMP0078 NEW)
endif()
```

Reply:
Done

Reply:
Done

Reply:
It seems you have added `CMP0087` first and later changed it to `CMP0078`?

Anyway, if your changes require the new behavior for this policy (namely, generator expressions support for `install`), wouldn't this break things for earlier versions of CMake which don't have support for this feature included?

Reply:
In my change I've kept it as CMP0087 - expecting that 78 was a typo intended to be 87.

If I remember correctly, I only added that to prevent a warning so might not be critical.

Reply:
@dmitrio95 Thanks for spotting this! My fault. I swapped the numbers in my comment !

I agree that the NEW behavior should only be enforced when it does not break things. As usual, CMake documentation is rubbish: no explanation of the problem that this new behavior solves, no examples, no tips for changes to do in cmake scripts, ... just template garbage to fill four paragraphs. And by googling for CMP0087 I did not find  anything relevant.

My conclusion was that generator expressions (as defined here https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html ) are now accepted and evaluated in `install()` sentences. But, as lomse cmake script does not use generator expressions in install commands, it is irrelevant what behavior is used. Thus, as the old behavior is deprecated, it would be better to enable the new behavior to have the script better prepared in case of future changes in `install()` statements. And in any case, to test the PR before merging, to check that nothing is broken.

Any thoughts?

Reply:
> In my change I've kept it as CMP0087 - expecting that 78 was a typo intended to be 87.

The number is correct in the commit message but still seems to be reverted in the change itself: https://github.com/lenmus/lomse/pull/270/commits/5f7cb5c26524e36c1ee8e796dda9a27a76f84cba

> as lomse cmake script does not use generator expressions in install commands, it is irrelevant what behavior is used

The warning is probably a result of the change made it this PR which does add a generator expression to an `install` command (see [these lines](https://github.com/lenmus/lomse/pull/270/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR887-R890)). If I understand the documentation [here](https://cmake.org/cmake/help/latest/policy/CMP0087.html) correctly, this feature is only supported since CMake 3.14 so earlier versions of CMake may be unable to handle this correctly. That's why I was asking about this: the change will probably work with newer versions of CMake but may lead to unexpected results with versions before 3.14.

Reply:
My apologies, you're absolutely right. I had fully intended it to be CMP0087 but seems that I've accidentally copy/pasted it as 78 from Cecilio's comment above.

Creating a symbolic link to the .so wasn't working, hence why I changed it to 
```
ln -fs ${CMAKE_INSTALL_PREFIX}/lib/$<TARGET_FILE_NAME:lomse-shared> 
                                        ${CMAKE_INSTALL_PREFIX}/lib/liblomse.so
```
which then worked correctly.
Really sorry but I'll have to leave this one with you to figure out - or revert the change, if that's necessary (although it took me a long time to figure out why linking to the shared library was failing!)

Reply:
@dmitrio95 You are right! I overlooked that a generator expression was introduced. I'm going to test this PR with cmake < 3.14 and also investigate the issue of the symlink commented by @grantsloman

Reply:
PR tested. When using cmake version 3.10.2, as expected, the generator expression is not evaluated and the resulting command is
```
ln -fs /usr/local/lib/$<TARGET_FILE_NAME:lomse-shared> /usr/lib/liblomse.so
```
and fails to create the symlink. So it is clear that CMP0087 NEW behavior is not acceptable unless forcing to require a cmake >= 3.14

@grantsloman To try understand why the unpatched `CMakeLists.txt` failed to create the symlink in macOS it would be useful to look at the generated command. And this is simple because in build folder cmake generates a file named `"cmake_install.cmake"`. Please open it with a text editor and look for string `"EXECUTE_PROCESS(COMMAND ln -fs"`. There is only one occurrence, around line 78, e.g. (when using cmake 3.10.2):
```
  EXECUTE_PROCESS(COMMAND ln -fs /usr/local/lib/$<TARGET_FILE_NAME:lomse-shared> 
                                        /usr/local/lib/liblomse.so )
```
There you will see the `ln` command and its arguments and you can try to find why it does not work for you.

Reply:
@grantsloman
A posible solution to supress `fatal: No names found, cannot describe anything. ` message is to add `ERROR_QUIET` parameter to `execute_command`:
```
    execute_process(COMMAND "${GIT_EXECUTABLE}" describe --tags --long --dirty
      WORKING_DIRECTORY "${LOMSE_ROOT_DIR}"
      OUTPUT_VARIABLE LOMSE_VERSION_GIT
      OUTPUT_STRIP_TRAILING_WHITESPACE
      ERROR_QUIET)
```
I have tested and it works.

Reply:
If I change the CMakeLists.txt line back to
```
"EXECUTE_PROCESS(COMMAND ln -fs ${CMAKE_INSTALL_PREFIX}/lib/${LOMSE_SHARED} 
                                        ${CMAKE_INSTALL_PREFIX}/lib/liblomse.so )" 
```
then the resulting generated command in `cmake_install.cmake` is
```
EXECUTE_PROCESS(COMMAND ln -fs /usr/local/lib/lomse-shared 
                                        /usr/local/lib/liblomse.so )
```
which isn't the correct target file name.

Reply:
I've done some tests and, curiously, the symlink is always created even when removing these lines:
```
    #create symlink to library
	install( CODE
        "EXECUTE_PROCESS(COMMAND ln -fs ${CMAKE_INSTALL_PREFIX}/lib/${LOMSE_SHARED} 
                                        ${CMAKE_INSTALL_PREFIX}/lib/liblomse.so )" 
    )
```

Therefore it seems irrelevant if that command is correct or not, and can be removed.

It looks as if cmake automatically creates the library and the symlink at build time. You can find them in build folder. Please notice that the previous command does not create a symlink in build folder so it is automatically created by cmake. And both are copied to `/usr/local/lib` at install time, and the symlink is adjusted to point to the installed library.

I have tested removing the previous lines from CMakeLists.txt and it works perfectly and the symlink is created at installation time.

I'm not been able to find any cmake documentation explaining this and confirming that symlinks are automatically created for libraries, but my tests suggest this. I am missing something?

@grantsloman Could you please test this, removing previous lines, in macOS?